lib/matplotlib/widgets.py

```diff
@@ -3267,97 +3338,120 @@ def corners(self):
+                    np.array([y[0], y[1], y[1], y[0]]))
 
     @property
     def edge_centers(self):
         """
         Midpoint of rectangle edges in data coordiantes from left,
         moving anti-clockwise.
         """
-        x0, y0, width, height = self._rect_bbox
-        w = width / 2.
-        h = height / 2.
-        xe = x0, x0 + w, x0 + width, x0 + w
-        ye = y0 + h, y0, y0 + h, y0 + height
-        transform = self._get_rotation_transform()
-        coords = transform.transform(np.array([xe, ye]).T).T
-        return coords[0], coords[1]
+        c = self._selection_artist._get_edge_midpoints()
+        c = self.ax.transData.inverted().transform(c)
+        return c[:, 0], c[:, 1]
 
     @property
     def center(self):
         """Center of rectangle in data coordinates."""
-        x0, y0, width, height = self._rect_bbox
-        return x0 + width / 2., y0 + height / 2.
+        c = self._selection_artist.get_center()
+        # Convert from display to data coordinates
+        return self.ax.transData.inverted().transform(c)
 
     @property
     def extents(self):
         """
         Return (xmin, xmax, ymin, ymax) in data coordinates as defined by the
         bounding box before rotation.
         """
-        x0, y0, width, height = self._rect_bbox
-        xmin, xmax = sorted([x0, x0 + width])
-        ymin, ymax = sorted([y0, y0 + height])
-        return xmin, xmax, ymin, ymax
+        cx, cy = self.corners
+        return cx[0], cx[2], cy[0], cy[2]
 
     @extents.setter
     def extents(self, extents):
+        # Convert from data to figure coordinates
+        corner_min = self.ax.transData.transform((extents[0], extents[2]))
+        corner_max = self.ax.transData.transform((extents[1], extents[3]))
         # Update displayed shape
-        self._draw_shape(extents)
-        if self._interactive:
-            # Update displayed handles
-            self._corner_handles.set_data(*self.corners)
-            self._edge_handles.set_data(*self.edge_centers)
-            self._center_handle.set_data(*self.center)
+        self._draw_shape((corner_min[0], corner_max[0],
+                          corner_min[1], corner_max[1]))
         self.set_visible(self.visible)
         self.update()
 
     @property
     def rotation(self):
         """
-        Rotation in degree in interval [-45°, 45°]. The rotation is limited in
-        range to keep the implementation simple.
+        Rotation in degrees.
         """
         return np.rad2deg(self._rotation)
 
     @rotation.setter
     def rotation(self, value):
-        # Restrict to a limited range of rotation [-45°, 45°] to avoid changing
-        # order of handles
-        if -45 <= value and value <= 45:
-            self._rotation = np.deg2rad(value)
-            # call extents setter to draw shape and update handles positions
-            self.extents = self.extents
+        self._rotation = np.deg2rad(value)
```

Comment:
I tried this, but it gets a bit confusing. Because the transform order of a `Rectangle` is

1. Translate the lower left corner to (x0, y0)
2. Rotate

If the rotation is not about (x0, y0), the lower left corner ends up at (x, y) != (x0, y0), which makes for a confusing mess in various bits of other logic.